### DISCUSSION TITLE
`mc mirror -watch` is not working when object deleted from source

### DISCUSSION THREAD

**Discussion by AhmedAbouelkher** [minio/minio#16132] [Q&A]
I'm trying to mirror my minio deployment bucket (source) with my AWS S3 bucket (target).

Object creation mirroring works great, when deleting an object, I noticed it wasn't deleted from my s3 bucket to _mirror_ the origin.
```bash
mc mirror --watch myminio/tst s3/rraw-media-upload
```
Checked the target bucket permissions with no luck.

Tried to run with `--remove` flag
```bash
mc mirror --watch --remove myminio/tst s3/rraw-media-upload
```
it deleted the target files which don't exist in the source but I need to run the command again after a source delete to delete the unwanted object on the target, which is not practical at all.

I need to be able to sync both object creation, deletion and update.

**Comment by harshavardhana** [comment]
Please share --debug output for the entire run

**Comment by AhmedAbouelkher** [comment]
Uploaded `screen.png` to source using this command

```bash
mc mirror --watch myminio/tst s3/rraw-media-upload --remove --debug
```

![Screen Shot 2022-11-27 at 11 31 36 PM](https://user-images.githubusercontent.com/56558577/204160729-c6e5baca-28dc-45dd-a446-e15b65935c88.png)

When removing `screen.png` there are no logs at all from the `mc` command

**Comment by AhmedAbouelkher** [comment]
@harshavardhana 
Any news? I still have the same issue

**Comment by aagusfernandez02** [comment]
@AhmedAbouelkher hi, could you solve it? I'm having the same problem
